flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -119,9 +141,45 @@ public static HiveObjectConversion getConversion(ObjectInspector inspector) {
 				} else {
 					return IdentityConversion.INSTANCE;
 				}
+			} else if (inspector instanceof JavaDateObjectInspector) {
+				if (((JavaDateObjectInspector) inspector).preferWritable()) {
+					return o -> new DateWritable((Date) o);
+				} else {
+					return IdentityConversion.INSTANCE;
+				}
+			} else if (inspector instanceof JavaTimestampObjectInspector) {
+				if (((JavaTimestampObjectInspector) inspector).preferWritable()) {
+					return o -> new TimestampWritable((Timestamp) o);
+				} else {
+					return IdentityConversion.INSTANCE;
+				}
+			} else if (inspector instanceof JavaBinaryObjectInspector) {
+				if (((JavaBinaryObjectInspector) inspector).preferWritable()) {
+					return o -> new BytesWritable((byte[]) o);
+				} else {
+					return IdentityConversion.INSTANCE;
+				}
+			} else if (inspector instanceof JavaHiveCharObjectInspector) {
+				if (((JavaHiveCharObjectInspector) inspector).preferWritable()) {
+					return o -> new HiveCharWritable(
+						new HiveChar((String) o, ((CharType) dataType.getLogicalType()).getLength()));
+				} else {
+					return IdentityConversion.INSTANCE;
```

Comment:
If o is a String (as the code shows), then identity conversion might not be expected because for char(p) type, Hive expects a HiveChar object instead of a string. However, I'm not 100% sure.

Same for varchar.

Reply:
You are right, I think it should be HiveChar instead